UVicMartletplace/martletplace
# Gavin/115 create api endpoint stubs for recommender

# Description
This PR includes the stub methods/endpoints to implement for the recommender API.

Additionally, it includes some definitions of BaseModel classes which help to clearly define the allowed inputs and (successful response) outputs.

Closes #115 

## How to Test
1. Run 'docker-compose up --build' in the root of the martletplace repo.
2. Enter the following into your browser:
http://localhost:8222/api/recommendations?authorization=spaghetti&page=1&limit=20
- Your output should look like this:
![image](https://github.com/UVicMartletplace/martletplace/assets/90437591/cc30463a-eb39-4c5f-ab5e-8091bd90b444)

3. Enter the following into your browser:
http://localhost:8222/docs
- The page should look like this:
![image](https://github.com/UVicMartletplace/martletplace/assets/90437591/5cb7db70-9478-48fa-98bc-080d45cd7bb9)
4. For each endpoint shown, click 'try it out', fill out the parameter fields, and execute. Check the outputs.


## Checklist
